protocol/v2/qbft/spectest/qbft_mapping_test.go

```diff
@@ -108,8 +108,11 @@ func TestQBFTMapping(t *testing.T) {
 			err = pre.Decode(preByts)
 			require.NoError(t, err)
 			typedTest.Pre = pre
+			t.Run(typedTest.Name, func(t *testing.T) { // using only spec struct so no need to run our version (TODO: check how we choose leader)
+				t.Parallel()
+				RunTimeout(t, typedTest)
+			})
```

Comment:
I know `t.parallel()` pattern is used everywhere. Any idea why we are not simply doing it once on the first line of QBFT Mapping? If there is no reason maybe we can make the code better if @MatusKysel agrees

I don't understand the TODO... leader is chosen in round robin

Reply:
@GalRogozinski TODOs  are here before my changes. I just left this as it was

Reply:
`t.parallel()` -- should be called inside every `t.Run()` calling it once outside the for loop makes the tests run slower. Same for `ssv_mapping_test.go`.  Tests this 

About the TODOs. I left it as it was before